crates/solver/src/order_balance_filter.rs

```diff
@@ -124,6 +129,24 @@ pub fn balance_orders(
     result
 }
 
+/// Prioritise which orders to account first for using users remaining balance.
+/// Given the external price vector, orders are sorted descending by the
+/// expected surplus (likelyhood of being matchable)
+fn sort_orders_for_balance_priority(orders: &mut [Order], external_prices: &ExternalPrices) {
```

Comment:
This probably needs to be ported to the co-located driver. Can be done in a separate PR though.

Reply:
@cowfee please correct me if I'm wrong but from what I can tell the colocated driver doesn't filter any orders based on remaining balances (https://github.com/cowprotocol/services/blob/ed60fd55f45696281213caa9e2d0c4fc792d0d51/crates/driver/src/infra/solver/dto/auction.rs#L40-L43)? Is this handled in each solver individually?

Reply:
Ahh, we don't have this in colocation. The issue about this is https://github.com/cowprotocol/services/issues/1408 .